src/main/java/org/jenkins/plugins/lockableresources/LockableResource.java

```diff
@@ -216,8 +296,44 @@ private void validateQueuingTimeout() {
 		}
 	}
 
+	public void queueAdd(StepContext context) {
+		queuedContexts.add(context);
+	}
+
+	/**
+	 * It removes the returned context from the queue and set the holding build (@link {@link #build}) to null
+	 * if there are no more contexts for that build.
+	 */
+	@CheckForNull
+	public StepContext getNextQueuedContext(boolean inversePrecedence) {
+		if (queuedContexts.size() > 0) {
+			if (!inversePrecedence) {
+				return queuedContexts.remove(0);
+			} else {
+				int newest = 0;
+				int index = 0;
+				int newestIndex = 0;
+				for (StepContext c : queuedContexts) {
+					try {
+						Run<?, ?> run = c.get(Run.class);
+						if (run.getNumber() > newest) {
```

Comment:
What if `run` is from a different `Job` than the one represented by `newest`? Then the build number comparison will be meaningles.